src/client/hud.cpp

```diff
@@ -693,6 +693,16 @@ void drawItemStack(video::IVideoDriver *driver,
 		driver->setTransform(video::ETS_VIEW, oldViewMat);
 		driver->setTransform(video::ETS_PROJECTION, oldProjMat);
 		driver->setViewPort(oldViewPort);
+
+		// draw the inventory_overlay
+		if (def.type == ITEM_NODE && def.inventory_image.empty() &&
+				!def.inventory_overlay.empty()) {
+			ITextureSource *tsrc = client->getTextureSource();
+			video::ITexture *overlay_texture = tsrc->getTexture(def.inventory_overlay);
+			core::dimension2d<u32> dimens = overlay_texture->getOriginalSize();
```

Comment:
make sure that `getTexture()` doesn't grab the texture and thus require the user to `->drop()` the texture

Reply:
I think, I'm not sure what you mean.

Reply:
[`getTexture`](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L645) [uses](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L647) [`getTextureId`](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L463) [which uses](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L484) [`generateTexture`](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L558) [which uses](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L601) [which uses](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L601) [from the class `video::IVideoDriver`](https://github.com/minetest/minetest/blob/42e1a127140965cac1be6e51e48192e341c2a29e/src/client/tile.cpp#L589) [`irr::video::IVideoDriver::addTexture`](http://irrlicht.sourceforge.net/docu/classirr_1_1video_1_1_i_video_driver.html#abfebeb09a692c0d6b4741d952d97668e) which
>Returns:
    Pointer to the newly created texture. This pointer should not be dropped.

. (However, this function is virtual, ergo that information might be not true, but this would be extremely weird.)
Ergo the texture doesn't need to be dropped, I think.

Reply:
Also in other places (eg. [here](https://github.com/minetest/minetest/blob/6cb6aea969032e43395ddbad09233e7e20830909/src/client/hud.cpp#L244-L248)) the texture isn't dropped, too.